workitem/expression_compiler_blackbox_test.go

```diff
@@ -200,6 +200,25 @@ func TestIsNull(t *testing.T) {
 	expect(t, c.IsNull("SpaceID"), `(`+workitem.Column(wiTbl, "space_id")+` IS NULL)`, []interface{}{}, nil)
 }
 
+func TestChild(t *testing.T) {
+	t.Parallel()
+	resource.Require(t, resource.UnitTest)
+	defJoins := workitem.DefaultTableJoins()
+	t.Run("iteration", func(t *testing.T) {
+		j := *defJoins["iteration"]
+		j.Active = true
+		e := "(uuid(\"work_items\".fields->>'system.iteration') IN (\n\t\t\t\tSELECT iter.id\n\t\t\t\t\tWHERE\n\t\t\t\t\t\t(SELECT j.path\n\t\t\t\t\t\t\tFROM iterations j\n\t\t\t\t\t\t\tWHERE j.id = ?\n\t\t\t\t\t\t) @> iter.path\n\t\t\t\t\t\t\t  ))"
+		expect(t, c.Child(c.Field("system.iteration"), c.Literal("abcd")), e, []interface{}{"abcd"}, []*workitem.TableJoin{&j})
+	})
+	t.Run("area", func(t *testing.T) {
```

Comment:
You're only testing for the allowed fields here but you have an error handling inside your code that does check for invalid field names. Therefore please also test with `system.label` for example.

Reply:
Done. 7e29206d2c045c15da9369317bca8031970575ce